shared/core/types/relay.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+var (
+	errInvalidRelayInvalidPayload  = errors.New("invalid relay payload")
+	errInvalidJSONRPCInvalidRPC    = errors.New("invalid value for JSONRPC field")
+	errInvalidJSONRPCMissingMethod = errors.New("Method field not set")
+	errInvalidRESTPayload          = errors.New("invalid REST payload")
+)
+
+// INCOMPLETE: perform any possible metadata validation
+// Validate performs validation on the relay payload
+func (r Relay) Validate() error {
+	if jsonRpcPayload := r.GetJsonRpcPayload(); jsonRpcPayload != nil {
+		return jsonRpcPayload.Validate()
+	}
+
+	if jsonPayload := r.GetRestPayload(); jsonPayload != nil {
+		return jsonPayload.Validate()
+	}
+
+	return fmt.Errorf("%w: %v", errInvalidRelayInvalidPayload, r)
+}
+
+// Validate performs validation on JSONRPC payload. More specifically, it verifies that:
+//  1. The JSONRPC field is set to "2.0" as per the JSONRPC spec requirement, and
+//  2. The Method field is not empty
+func (p JSONRPCPayload) Validate() error {
+	if p.JsonRpc != "2.0" {
+		return fmt.Errorf("%w: %s", errInvalidJSONRPCInvalidRPC, p.JsonRpc)
+	}
+
+	// DISCUSS: do we need/want chain-specific validation? Potential for reusing the existing logic of Portal V2/pocket-go
```

Comment:
We still haven't fully defined what relay chains will look like.

For example, what if they're `JWT` like tokens/strings that can hold more data to support general purpose RPC? `<Type>.<ID>.<Config>`. See slide 26 form https://docs.google.com/presentation/d/19NqPV0e6JbQzzwqEnfJ_WiMqpjaPwWo-qui6Xr3pNSQ/edit#slide=id.g23e8b9d22f1_0_134.

I think updating the TODO with some extra details and leaving as is should be good.

Reply:
Thank you for the review. Fixed.